docker/context.go

```diff
@@ -0,0 +1,444 @@
+package docker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// ContextInfo represents a Docker context with its metadata
+type ContextInfo struct {
+	Name        string
+	Current     bool
+	Description string
+	DockerHost  string
+	TLS         bool
+	Error       string
+}
+
+// ListContexts returns all available Docker contexts using docker CLI
+func ListContexts() ([]ContextInfo, error) {
```

Comment:
Can you please ask if this function is idiomatic? This looks much more complex than necessary. Please always ask for generated code to be: idiomatic, according to best practice and strongly typed.